xdsl/ir/core.py

```diff
@@ -769,6 +769,13 @@ def walk_reverse(self) -> Iterator[Operation]:
             yield from region.walk_reverse()
         yield self
 
+    def get_attr_or_prop(self, name: str) -> Attribute | None:
```

Comment:
Seems like this isn't used? I feel like we should either remove this diff or use it and test it

Reply:
It is at https://github.com/xdslproject/xdsl/blob/63cd884c75dbd96af3dc7b965ef16e13f3275970/xdsl/interpreters/experimental/pdl.py#L120 :slightly_smiling_face:

Reply:
Could you add a test for the `return None`-path? Also, do we actually want to return None in that case or just throw an error?

Reply:
I see, None is used as the way to identify it's not in there for the interpreter. I personally think it makes more sense to throw en error here. Or maybe you could add a keyword argument to define what should be returned by default? (like list.get does)

Reply:
I think we need tests for all paths, and a clear documentation for why this is needed and whether it's temporary. It seems pretty weird to me.

Reply:
I'll add tests, and also doumentation for it yes!
The reason we want this is in order to be compatible with MLIR, we often need to get an attribute that might also be a property at that point.
I feel that this will stay for a while (especially since PDL uses it), but it might be removed once everything move to properties.
Note that MLIR uses *a lot* this method.